### PR TITLE
Loki: Added support to show label types in Log Details

### PR DESCRIPTION
**Pull request by matyax**
This PR proposes a method to display the type of each label in Log Details, for data sources that support label types such as Loki.

Demo:

### CODE REVIEW

File: public/app/features/logs/utils.ts
Comment (gtk-grafana) [inline public/app/features/logs/utils.ts:373]:
```suggestion
          return 'Structured metadata';
```

**Review by gtk-grafana — APPROVED** [review]
Yes please! Looks good

**Comment by matyax** [comment]
@svennergr What do you think? This one or https://github.com/grafana/grafana/pull/97334 ?

**Comment by gtk-grafana** [comment]
Would be nice if we had a different icon for each field type

**Comment by matyax** [comment]
Yeah, that'd be pretty cool. Maybe we can hack one with a circled small letter 🤔

**Comment by matyax** [comment]
We can do something like this:

imagen

**Comment by zizzpudding** [comment]
I'm curious what use case we're trying to solve here? 
Showing an icon seems to be emphasizing the difference between the fields, rather than the opposite. 

Could we do it in a more subtle way, so the information is there if you need it be not in your face. 

For instance with colors or font decorations?  Or with Grouping
image

image

**Comment by zizzpudding** [comment]
How about if we add a little separation between the actions and the indicator, like this
image

**Comment by matyax** [comment]
Love that, let's give it a try!

**Comment by matyax** [comment]
Added space and updated the demo. Let me know what you think 🧐

**Comment by liliwanggithub** [comment]
I can't wait for this, We are block for this for so long time, when it can be release ?  @matyax  @zizzpudding

**Comment by matyax** [comment]
@liliwanggithub We're just waiting for a team decision. Once it's made this should be merged and available soon after in Grafana Cloud and released by the end of January.

**Comment by gtk-grafana** [comment]
Be good to add this to the changelog, and update the milestone to 11.5

File: public/app/features/logs/utils.ts
Comment (svennergr) [inline public/app/features/logs/utils.ts:374]:
```suggestion
  return getDataSourceLabelType(labelType, row.datasourceType);
}

function getDataSourceLabelType(labelType: string, datasourceType: string) {
```

Just a nit, but `resolve` sounds too much like a `Promise` in my opinion :D

**Review by svennergr — APPROVED** [review]
LGTM, left a few nits.

In general I also agree with @zizzpudding:
> Could we do it in a more subtle way, so the information is there if you need it be not in your face.

But let's go with this for now.

**Pull request merged**
